book/chrome.md

```diff
@@ -1209,3 +1209,23 @@ addition to tabs. This will require keeping track of multiple Tk
 windows and canvases and grouping tabs by their containing window.
 You'll also need some way to create a new window, perhaps with a
 keypress such as `Ctrl+N`.
+
+*Hit testing in layout*: The `click` method we implemented is on the `Tab`
+object. While it doesn't have a whole lot of logic in it, there is special
+logic for `Text` objects and `a` tags. Real browsers have many more special
+kinds of nodes, plus more complicated layout, so they tend to implement this
+logic directly on the layout tree.^[And they also use this logic for more than
+just clicking. The more general task of determining which objects on screen
+overlap a given point or region is called *hit testing*---by analogy with
+whether an arrow shot at that location would "hit" the object.] Implement
+`click` on the layout tree.
+
+*Hit testing on the display list*: Hit testing can be thought of as a "reversed"
+version of `paint`, since it looks at the elements front-to-back in paint order,
+as opposed to back-to-front. Building on this observation, we could also create
+a special display list that represents hit testing information, or build all
+of the necessary information for hit testing directly into the display list
+instead of the layout tree. Implement one of these.^[You might want to implement
+hit testing in this way in a browser because display lists are pure data
+structurs and therefore easier to optimize or execute in different
+threads.]
```

Comment:
```suggestion
instead of the layout tree. Implement one of these.^[This is more
challenging in real browsers, because you might want some kinds of
invisible elements to be clickable. But the advantage is that display
lists are pure data structures which are easy to pass between threads.]
```